.github/workflows/check-dist.yml

```diff
@@ -32,8 +32,8 @@ jobs:
         with:
           node-version: 16.x
           cache: npm
-      - name: Install dependencies
-        run: npm install
+      - name: remove js files in dist/
+        run: find dist/ -type f \( -name "*.json" -o -name "*.js" -o -name "*.js.map" \) -delete
```

Comment:
```suggestion
```
1. Correct, we could just remove the `Install dependencies` step because `node_modules` is already checked in.
2. Regarding the "bug" you found, the fix does not work. What if the manually added file is `hello.txt`? It won't be detected either.
3. I would say it's not a bug for `check-dist`. The goal for `check-dist` is to detect differences caused by forgetting of running `npm run build`. In other words, `if and only if` files created by `npm run build` matches each other, then `check-dist` should be deemed as `passed`. Other items, such as manually added files SHOULD be detected by code reviewers of the PR.

Reply:
> What if the manually added file is hello.txt? It won't be detected either.

IMO, the purpose of `check-dist` job is to make sure that only build results of `npm run build` is checked into `dist\` folder. The build won't generate `hello.txt` so it's out of scope for detection.

>  In other words, if and only if files created by npm run build matches each other, then check-dist should be deemed as passed.

Correct, but build results from **previous** `npm build run` should not be checked into `dist\` folder, that's what currently `check-dist` cannot do. E.g. if I have a `dist\a.js` which is checked in with it's correspondent typescript file `src/a.ts` after running `npm run build`. The `check-dist` job should succeed. Based on that, if I checked out `src/a.ts` and run `npm run build` again, the file `dist\a.js` is still in there and `check-dist` job still succeeds.

Reply:
@Two-Hearts If you think old build result cleaning in the `dist\` folder is not necessary, please let me know. I can remove that part from this PR.

Reply:
> Based on that, if I checked out `src/a.ts` and run `npm run build` again, the file `dist\a.js` is still in there and `check-dist` job still succeeds.

@qweeah Here's the official `check-dist` of GitHub Actions: https://github.com/actions/reusable-workflows/blob/main/.github/workflows/check-dist.yml, please take a look.
If removing the source file and run `npm run build` at local does not remove the built file from `dist`, then `check-dist` is designed to be `pass` in this case. `check-dist` checks exactly what `npm run build` does to the repo. 
For a PR review, removing a source file is a big change to the repo, the corresponding built file is expected to be removed and validated by `code reviewers`/`repo maintainers`.

Reply:
Got it, I will remove the part that cleans the old build.